tracer/src/Datadog.Trace/FeatureFlags/Exposure/ExposureApi.cs

```diff
@@ -0,0 +1,163 @@
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Datadog.Trace.Agent;
+using Datadog.Trace.Agent.Transports;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.FeatureFlags.Exposure.Model;
+using Datadog.Trace.HttpOverStreams;
+using Datadog.Trace.Logging;
+using Datadog.Trace.Vendors.Newtonsoft.Json;
+
+namespace Datadog.Trace.Exposure
+{
+    internal sealed class ExposureApi : IDisposable
+    {
+        internal static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(ExposureApi));
+
+        public const string ExposurePath = "api/v2/exposure";
+        private readonly TaskCompletionSource<bool> _processExit = new();
+        private readonly TimeSpan _sendInterval = TimeSpan.FromSeconds(10);
+        private readonly Queue<ExposureEvent> _exposures = new Queue<ExposureEvent>();
+        private IApiRequestFactory? _apiRequestFactory = null;
+        private Dictionary<string, string> _context = new Dictionary<string, string>();
+        private int _started = 0;
+
+        private ExposureApi(TracerSettings tracerSettings)
+        {
+            UpdateSettings(tracerSettings.Manager.InitialExporterSettings, tracerSettings.Manager.InitialMutableSettings);
+            tracerSettings.Manager.SubscribeToChanges(settingChanges => UpdateSettings(settingChanges.UpdatedExporter ?? settingChanges.PreviousExporter, settingChanges.UpdatedMutable ?? settingChanges.PreviousMutable));
+        }
+
+        private void UpdateSettings(ExporterSettings exporterSettings, MutableSettings settings)
+        {
+            Log.Debug("ExposureApi::UpdateSettings -> Applyting settings");
+            var apiRequestFactory = AgentTransportStrategy.Get(
+                exporterSettings,
+                productName: "FeatureFlags exposure",
+                tcpTimeout: TimeSpan.FromSeconds(5),
+                AgentHttpHeaderNames.MinimalHeaders,
+                () => new MinimalAgentHeaderHelper(),
+                uri => uri);
+            Interlocked.Exchange(ref _apiRequestFactory, apiRequestFactory);
+
+            var context = new Dictionary<string, string>
+            {
+                { "service", settings.ServiceName ?? "unknown" },
+                { "env", settings.Environment ?? "unknown" },
+                { "version", settings.ServiceVersion ?? "unknown" }
+            };
+            Interlocked.Exchange(ref _context, context);
+        }
+
+        public static ExposureApi Create(TracerSettings tracerSettings)
+        {
+            return new ExposureApi(tracerSettings);
+        }
+
+        public void Dispose()
+        {
+            try { _processExit.TrySetResult(true); }
+            catch { }
+        }
```

Comment:
meh, this shouldn't throw 😄 

```suggestion
        public void Dispose()
        {
            _processExit.TrySetResult(true);
        }
```

Reply:
Changed